<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>Ed Brindley</author>
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>950</width>
    <height>688</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>BadgerComputer</string>
  </property>
  <property name="unifiedTitleAndToolBarOnMac">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QPlainTextEdit" name="plainTextEdit"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>950</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>Fi&amp;le</string>
    </property>
    <widget class="QMenu" name="menuRecent">
     <property name="title">
      <string>&amp;Recent</string>
     </property>
     <property name="icon">
      <iconset resource="resources/app.qrc">
       <normaloff>:/toolbar/document-open-recent.svg</normaloff>:/toolbar/document-open-recent.svg</iconset>
     </property>
    </widget>
    <addaction name="actionLoad_ROM"/>
    <addaction name="menuRecent"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menuEmu_Control">
    <property name="title">
     <string>Em&amp;u Control</string>
    </property>
    <addaction name="actionRun"/>
    <addaction name="actionStop"/>
    <addaction name="actionReset"/>
    <addaction name="actionStep"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionScroll_memory_with_PC"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEmu_Control"/>
   <addaction name="menuView"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QDockWidget" name="dockWidget">
   <property name="windowTitle">
    <string>Re&amp;gisters</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QGridLayout" name="gridLayout">
     <item row="8" column="1">
      <widget class="QLabel" name="a_value">
       <property name="text">
        <string>0</string>
       </property>
      </widget>
     </item>
     <item row="11" column="0">
      <widget class="QLabel" name="d_label">
       <property name="text">
        <string>D</string>
       </property>
      </widget>
     </item>
     <item row="10" column="1">
      <widget class="QLabel" name="c_value">
       <property name="text">
        <string>0</string>
       </property>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="sp_label">
       <property name="text">
        <string>SP</string>
       </property>
      </widget>
     </item>
     <item row="5" column="1">
      <widget class="QLabel" name="ix_value">
       <property name="text">
        <string>0</string>
       </property>
      </widget>
     </item>
     <item row="11" column="1">
      <widget class="QLabel" name="d_value">
       <property name="text">
        <string>0</string>
       </property>
      </widget>
     </item>
     <item row="6" column="0">
      <widget class="QLabel" name="iy_label">
       <property name="text">
        <string>IY</string>
       </property>
      </widget>
     </item>
     <item row="7" column="1">
      <widget class="QLabel" name="hl_value">
       <property name="text">
        <string>0</string>
       </property>
      </widget>
     </item>
     <item row="7" column="0">
      <widget class="QLabel" name="hl_label">
       <property name="text">
        <string>HL</string>
       </property>
      </widget>
     </item>
     <item row="5" column="0">
      <widget class="QLabel" name="ix_label">
       <property name="text">
        <string>IX</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLabel" name="pc_value">
       <property name="text">
        <string>0</string>
       </property>
      </widget>
     </item>
     <item row="10" column="0">
      <widget class="QLabel" name="c_label">
       <property name="text">
        <string>C</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="pc_label">
       <property name="text">
        <string>PC</string>
       </property>
      </widget>
     </item>
     <item row="9" column="1">
      <widget class="QLabel" name="b_value">
       <property name="text">
        <string>0</string>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QLabel" name="sp_value">
       <property name="text">
        <string>0</string>
       </property>
      </widget>
     </item>
     <item row="9" column="0">
      <widget class="QLabel" name="b_label">
       <property name="text">
        <string>B</string>
       </property>
      </widget>
     </item>
     <item row="13" column="0">
      <widget class="QLabel" name="f_label">
       <property name="text">
        <string>F</string>
       </property>
      </widget>
     </item>
     <item row="12" column="1">
      <widget class="QLabel" name="e_value">
       <property name="text">
        <string>0</string>
       </property>
      </widget>
     </item>
     <item row="6" column="1">
      <widget class="QLabel" name="iy_value">
       <property name="text">
        <string>0</string>
       </property>
      </widget>
     </item>
     <item row="8" column="0">
      <widget class="QLabel" name="a_label">
       <property name="text">
        <string>A</string>
       </property>
      </widget>
     </item>
     <item row="12" column="0">
      <widget class="QLabel" name="e_label">
       <property name="text">
        <string>E</string>
       </property>
      </widget>
     </item>
     <item row="13" column="1">
      <widget class="QLabel" name="f_value">
       <property name="text">
        <string>0</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="memoryDockWidget">
   <property name="minimumSize">
    <size>
     <width>196</width>
     <height>198</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>&amp;Memory disassembly</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="5" column="0" colspan="2">
      <widget class="QTableView" name="disassemblyView">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="font">
        <font>
         <pointsize>8</pointsize>
         <weight>75</weight>
         <bold>true</bold>
        </font>
       </property>
       <property name="alternatingRowColors">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionQuit"/>
   <addaction name="actionLoad_ROM"/>
   <addaction name="actionRun"/>
   <addaction name="actionStep"/>
   <addaction name="actionContinue"/>
   <addaction name="actionStop"/>
   <addaction name="actionDebug_stats"/>
   <addaction name="actionAbout"/>
  </widget>
  <action name="actionAbout">
   <property name="icon">
    <iconset resource="resources/app.qrc">
     <normaloff>:/toolbar/help-about.svg</normaloff>:/toolbar/help-about.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;About</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="icon">
    <iconset resource="resources/app.qrc">
     <normaloff>:/toolbar/application-exit.svg</normaloff>:/toolbar/application-exit.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Quit</string>
   </property>
  </action>
  <action name="actionLoad_ROM">
   <property name="icon">
    <iconset resource="resources/app.qrc">
     <normaloff>:/toolbar/document-open.svg</normaloff>:/toolbar/document-open.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Load ROM</string>
   </property>
  </action>
  <action name="actionRun">
   <property name="icon">
    <iconset resource="resources/app.qrc">
     <normaloff>:/toolbar/media-playback-start.svg</normaloff>:/toolbar/media-playback-start.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Run</string>
   </property>
  </action>
  <action name="actionStop">
   <property name="icon">
    <iconset resource="resources/app.qrc">
     <normaloff>:/toolbar/media-playback-stop.svg</normaloff>:/toolbar/media-playback-stop.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Pause</string>
   </property>
  </action>
  <action name="actionReset">
   <property name="text">
    <string>R&amp;eset</string>
   </property>
  </action>
  <action name="actionStep">
   <property name="icon">
    <iconset resource="resources/app.qrc">
     <normaloff>:/toolbar/arrow-right.svg</normaloff>:/toolbar/arrow-right.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Step</string>
   </property>
  </action>
  <action name="actionDebug_stats">
   <property name="text">
    <string>Debug stats</string>
   </property>
  </action>
  <action name="actionContinue">
   <property name="text">
    <string>Continue</string>
   </property>
  </action>
  <action name="actionScroll_memory_with_PC">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>&amp;Scroll memory with PC</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="resources/app.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionAbout</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>showAboutBox()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>474</x>
     <y>343</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLoad_ROM</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>loadRom()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>474</x>
     <y>343</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>quit()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>474</x>
     <y>343</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRun</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>run()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>474</x>
     <y>343</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionStop</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>stop()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>474</x>
     <y>343</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionReset</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>reset()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>474</x>
     <y>343</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionStep</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>step()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>474</x>
     <y>343</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionDebug_stats</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>print_debug_stats()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>474</x>
     <y>343</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionContinue</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>resume()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>474</x>
     <y>343</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionScroll_memory_with_PC</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>toggleScrollMemory(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>474</x>
     <y>343</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>loadRom()</slot>
  <slot>quit()</slot>
  <slot>showAboutBox()</slot>
  <slot>run()</slot>
  <slot>stop()</slot>
  <slot>reset()</slot>
  <slot>step()</slot>
  <slot>print_debug_stats()</slot>
  <slot>resume()</slot>
  <slot>toggleScrollMemory(bool)</slot>
 </slots>
</ui>
